### ISSUE TITLE
crontab not working inside container

### ISSUE THREAD

**Issue by megapearl** [jerheij/docker-spotweb#13]
Hi,

The crontab inside the container looks wrong and is not working:
I'm using your 'latest' master branch.

My docker-compose is:

```
  spotweb:
    image: jerheij/spotweb:latest
    container_name: spotweb
    environment:
      - PUID=1015
      - PGID=101
      - TZ=Europe/Amsterdam
      - SQL=mysql
      - SPOTWEB_CRON_RETRIEVE=*/15 * * * * root /usr/bin/php /var/www/spotweb/retrieve.php
    volumes:
      - /home/dockers/spotweb/dbsettings.inc.php:/var/www/spotweb/dbsettings.inc.php
      - /mnt/storage/Uploads/Watchdir:/watchdir
    ports:
      - 8080:80
    restart: unless-stopped
```

But inside the container it looks like:

```
bash-5.1# cat /etc/crontabs/root 
*/15 * * * * root /usr/bin/php /var/www/spotweb/retrieve.php /usr/bin/php /var/www/spotweb/retrieve.php >/var/log/stdout 2>&1
bash-5.1#
```
Why twice? what's wrong?

Regards,
Donald.

**Comment by megapearl** [comment]
Changed the  SPOTWEB_CRON_RETRIEVE env to just: `*/15 * * * *`
and that seems to work.
It was unclear for me in the documentation.

I do see some other errors but I don't know if they are related to your docker, spotweb or nzbget.
Also the time is -2 hours offset.

My ENV TZ=Europe/Amsterdam but it did't seem to work.

Regards,
Donald.

```
bash-5.1# cat error.log 
[Sat Aug 13 13:38:40.183737 2022] [mpm_prefork:notice] [pid 1] AH00163: Apache/2.4.53 (Unix) PHP/7.4.29 configured -- resuming normal operations
[Sat Aug 13 13:38:40.183819 2022] [core:notice] [pid 1] AH00094: Command line: '/usr/sbin/httpd -D FOREGROUND -f /etc/apache2/httpd.conf'
[Sat Aug 13 13:38:48.940618 2022] [php7:notice] [pid 33] [client 10.0.0.15:54422] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?page=index&search[tree]=cat0_z0_a3,cat0_z0_a10&search[value][]=
[Sat Aug 13 13:40:44.458899 2022] [php7:notice] [pid 34] [client 172.18.0.1:53668] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server
[Sat Aug 13 13:40:57.764421 2022] [php7:notice] [pid 35] [client 10.0.0.15:54466] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server
[Sat Aug 13 13:40:58.001833 2022] [php7:notice] [pid 35] [client 10.0.0.15:54466] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?page=index&search[tree]=cat0_z0_a3,cat0_z0_a10&search[value][]=
[Sat Aug 13 13:40:58.002768 2022] [php7:notice] [pid 36] [client 10.0.0.15:54467] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?page=index&search[tree]=cat0_z0_a3,cat0_z0_a10&search[value][]=
[Sat Aug 13 13:40:58.202234 2022] [php7:notice] [pid 36] [client 10.0.0.15:54467] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?page=index&search[tree]=cat0_z0_a3,cat0_z0_a10&search[value][]=
[Sat Aug 13 13:40:59.203767 2022] [php7:notice] [pid 36] [client 10.0.0.15:54467] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?page=index&search[tree]=cat0_z0_a3,cat0_z0_a10&search[value][]=
[Sat Aug 13 13:41:02.097473 2022] [php7:notice] [pid 36] [client 10.0.0.15:54467] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?page=index&search[tree]=cat0_z0_a3,cat0_z0_a10&search[value][]=
[Sat Aug 13 13:41:02.829414 2022] [php7:notice] [pid 35] [client 10.0.0.15:54466] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?page=index&search[tree]=cat0_z0_a3,cat0_z0_a10&search[value][]=
[Sat Aug 13 13:41:02.883450 2022] [php7:notice] [pid 37] [client 10.0.0.15:54469] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?page=index&search[tree]=cat0_z0_a3,cat0_z0_a10&search[value][]=
[Sat Aug 13 13:41:02.928683 2022] [php7:notice] [pid 39] [client 10.0.0.15:54470] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?page=index&search[tree]=cat0_z0_a3,cat0_z0_a10&search[value][]=
[Sat Aug 13 13:41:02.996556 2022] [php7:notice] [pid 33] [client 10.0.0.15:54471] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?page=index&search[tree]=cat0_z0_a3,cat0_z0_a10&search[value][]=
[Sat Aug 13 13:41:03.129403 2022] [php7:notice] [pid 34] [client 10.0.0.15:54472] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?page=index&search[tree]=cat0_z0_a3,cat0_z0_a10&search[value][]=
[Sat Aug 13 13:41:03.540915 2022] [php7:notice] [pid 36] [client 10.0.0.15:54467] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?page=index&search[tree]=cat0_z0_a3,cat0_z0_a10&search[value][]=
[Sat Aug 13 13:41:04.457035 2022] [php7:notice] [pid 35] [client 10.0.0.15:54466] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?page=index&search[tree]=cat0_z0_a3,cat0_z0_a10&search[value][]=
[Sat Aug 13 13:41:04.609269 2022] [php7:notice] [pid 37] [client 10.0.0.15:54469] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?page=index&search[tree]=cat0_z0_a3,cat0_z0_a10&search[value][]=
[Sat Aug 13 13:41:04.659685 2022] [php7:notice] [pid 33] [client 10.0.0.15:54471] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?page=index&search[tree]=cat0_z0_a3,cat0_z0_a10&search[value][]=
[Sat Aug 13 13:41:04.704659 2022] [php7:notice] [pid 39] [client 10.0.0.15:54470] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?page=index&search[tree]=cat0_z0_a3,cat0_z0_a10&search[value][]=
[Sat Aug 13 13:41:04.707280 2022] [php7:notice] [pid 34] [client 10.0.0.15:54472] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?page=index&search[tree]=cat0_z0_a3,cat0_z0_a10&search[value][]=
[Sat Aug 13 13:41:05.499576 2022] [php7:notice] [pid 36] [client 10.0.0.15:54467] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?page=index&search[tree]=cat0_z0_a3,cat0_z0_a10&search[value][]=
[Sat Aug 13 13:41:05.759603 2022] [php7:notice] [pid 35] [client 10.0.0.15:54466] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?page=index&search[tree]=cat0_z0_a3,cat0_z0_a10&search[value][]=
[Sat Aug 13 13:41:06.118238 2022] [php7:notice] [pid 33] [client 10.0.0.15:54471] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?page=index&search[tree]=cat0_z0_a3,cat0_z0_a10&search[value][]=
[Sat Aug 13 13:41:06.123137 2022] [php7:notice] [pid 37] [client 10.0.0.15:54469] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?page=index&search[tree]=cat0_z0_a3,cat0_z0_a10&search[value][]=
[Sat Aug 13 13:41:06.132375 2022] [php7:notice] [pid 39] [client 10.0.0.15:54470] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?page=index&search[tree]=cat0_z0_a3,cat0_z0_a10&search[value][]=
[Sat Aug 13 13:41:06.179611 2022] [php7:notice] [pid 34] [client 10.0.0.15:54472] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?page=index&search[tree]=cat0_z0_a3,cat0_z0_a10&search[value][]=
[Sat Aug 13 13:41:07.017386 2022] [php7:notice] [pid 36] [client 10.0.0.15:54467] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?page=index&search[tree]=cat0_z0_a3,cat0_z0_a10&search[value][]=
[Sat Aug 13 13:41:15.723999 2022] [php7:notice] [pid 62] [client 10.0.0.15:54477] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server
[Sat Aug 13 13:41:17.844069 2022] [php7:notice] [pid 62] [client 10.0.0.15:54477] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/
[Sat Aug 13 13:41:17.882132 2022] [php7:notice] [pid 71] [client 10.0.0.15:54478] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/
[Sat Aug 13 13:41:18.265743 2022] [php7:notice] [pid 72] [client 10.0.0.15:54479] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/
[Sat Aug 13 13:41:20.508058 2022] [php7:notice] [pid 72] [client 10.0.0.15:54479] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/
[Sat Aug 13 13:41:21.267704 2022] [php7:notice] [pid 72] [client 10.0.0.15:54479] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?search[tree]=&search[unfiltered]=true&search[value][]=New:0
[Sat Aug 13 13:41:21.298340 2022] [php7:notice] [pid 71] [client 10.0.0.15:54478] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?search[tree]=&search[unfiltered]=true&search[value][]=New:0
[Sat Aug 13 13:41:21.331607 2022] [php7:notice] [pid 62] [client 10.0.0.15:54477] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?search[tree]=&search[unfiltered]=true&search[value][]=New:0
[Sat Aug 13 13:41:21.350958 2022] [php7:notice] [pid 73] [client 10.0.0.15:54489] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?search[tree]=&search[unfiltered]=true&search[value][]=New:0
[Sat Aug 13 13:41:21.382198 2022] [php7:notice] [pid 74] [client 10.0.0.15:54490] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?search[tree]=&search[unfiltered]=true&search[value][]=New:0
[Sat Aug 13 13:41:21.413384 2022] [php7:notice] [pid 35] [client 10.0.0.15:54491] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?search[tree]=&search[unfiltered]=true&search[value][]=New:0
[Sat Aug 13 13:41:22.860243 2022] [php7:notice] [pid 35] [client 10.0.0.15:54491] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?search[tree]=&search[unfiltered]=true&search[value][]=New:0
[Sat Aug 13 13:41:22.864303 2022] [php7:notice] [pid 72] [client 10.0.0.15:54479] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?search[tree]=&search[unfiltered]=true&search[value][]=New:0
[Sat Aug 13 13:41:22.966001 2022] [php7:notice] [pid 62] [client 10.0.0.15:54477] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?search[tree]=&search[unfiltered]=true&search[value][]=New:0
[Sat Aug 13 13:41:22.979798 2022] [php7:notice] [pid 74] [client 10.0.0.15:54490] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?search[tree]=&search[unfiltered]=true&search[value][]=New:0
[Sat Aug 13 13:41:23.000050 2022] [php7:notice] [pid 71] [client 10.0.0.15:54478] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?search[tree]=&search[unfiltered]=true&search[value][]=New:0
[Sat Aug 13 13:41:23.150601 2022] [php7:notice] [pid 73] [client 10.0.0.15:54489] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?search[tree]=&search[unfiltered]=true&search[value][]=New:0
[Sat Aug 13 13:41:24.335426 2022] [php7:notice] [pid 71] [client 10.0.0.15:54478] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?search[tree]=&search[unfiltered]=true&search[value][]=New:0
[Sat Aug 13 13:41:24.387351 2022] [php7:notice] [pid 74] [client 10.0.0.15:54490] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?search[tree]=&search[unfiltered]=true&search[value][]=New:0
[Sat Aug 13 13:41:30.324115 2022] [php7:notice] [pid 74] [client 10.0.0.15:54490] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?search[tree]=&search[unfiltered]=true&search[value][]=New:0
[Sat Aug 13 13:41:53.130532 2022] [php7:notice] [pid 33] [client 10.0.0.15:54495] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?search[tree]=&search[unfiltered]=true&search[value][]=New:0
[Sat Aug 13 13:43:24.140581 2022] [php7:notice] [pid 37] [client 10.0.0.15:54509] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server
[Sat Aug 13 13:43:30.126734 2022] [php7:notice] [pid 39] [client 10.0.0.15:54513] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?search[tree]=&search[unfiltered]=true&search[value][]=New:0
[Sat Aug 13 13:43:31.462973 2022] [php7:notice] [pid 39] [client 10.0.0.15:54513] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?search[tree]=&search[unfiltered]=true
[Sat Aug 13 13:43:32.238304 2022] [php7:notice] [pid 39] [client 10.0.0.15:54513] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?page=index&search[tree]=cat0_z0&search[value][]=
[Sat Aug 13 13:43:32.788051 2022] [php7:notice] [pid 36] [client 10.0.0.15:54517] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?page=index&search[tree]=cat0_z0&search[value][]=
[Sat Aug 13 13:43:32.822019 2022] [php7:notice] [pid 73] [client 10.0.0.15:54518] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?page=index&search[tree]=cat0_z0&search[value][]=
[Sat Aug 13 13:43:32.840816 2022] [php7:notice] [pid 72] [client 10.0.0.15:54519] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?page=index&search[tree]=cat0_z0&search[value][]=
[Sat Aug 13 13:43:32.888528 2022] [php7:notice] [pid 62] [client 10.0.0.15:54520] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?page=index&search[tree]=cat0_z0&search[value][]=
[Sat Aug 13 13:43:32.905885 2022] [php7:notice] [pid 35] [client 10.0.0.15:54521] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?page=index&search[tree]=cat0_z0&search[value][]=
[Sat Aug 13 13:43:33.855215 2022] [php7:notice] [pid 39] [client 10.0.0.15:54513] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?page=index&search[tree]=cat0_z0&search[value][]=
[Sat Aug 13 13:43:34.117868 2022] [php7:notice] [pid 62] [client 10.0.0.15:54520] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?page=index&search[tree]=cat0_z0&search[value][]=
[Sat Aug 13 13:43:34.190672 2022] [php7:notice] [pid 72] [client 10.0.0.15:54519] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?page=index&search[tree]=cat0_z0&search[value][]=
[Sat Aug 13 13:43:38.188071 2022] [php7:notice] [pid 72] [client 10.0.0.15:54519] ERROR: Could not decode json-data for NZBGet method 'status', http returncode: 700 / Empty reply from server, referer: http://10.0.0.44:8080/?page=index&search[tree]=cat0_z0&search[value][]=
bash-5.1# time

real    0m0.000s
user    0m0.000s
sys     0m0.000s
bash-5.1# date
Sat Aug 13 13:44:34 UTC 2022
bash-5.1#
```

**Comment by jerheij** [comment]
Thanks for your feedback! I just had a look and the documentation regarding the crons could be a little bit more clear. I will update that somewhere in the upcoming week.

Have you tried putting quotes around the time zone?
`TZ="Europe/Amsterdam"`

Let me know if it help or if I need to look at the TZ functionality again!

Cheers!

**Comment by megapearl** [comment]
No, not yet, will try it now

**Comment by megapearl** [comment]
The quotes didn't make any difference... still a 2 hours offset.

Host is 16:13
Dockers is 14:13

For the NZBGet errors, it seems that is connecting unsecure to secure port, needed the check SSL and is working as expected again.

**Comment by jerheij** [comment]
So the only remaining issue is the timezone not working as expected?

**Comment by megapearl** [comment]
That's right.

**Comment by jerheij** [comment]
Okay, I'll have a look somewhere this week! 

I don't use Spotweb any more after moving out of the Netherlands so I don't notice those kind of issues. Decided, since I built some automation to keep the OS and the containers up to date, maintaining the Spotweb container is not that big of a deal and accepted the risk of not noticing all the issues right away!

I'll let you know once I find something or fixed it.

Thanks again!

**Comment by megapearl** [comment]
No problem, take your time :-)

**Comment by jerheij** [comment]
> No problem, take your time :-)

Should now be working! I expanded the timezone functionality so it applies to both PHP and the container itself. Let me know whether this works for you, it worked for my tests.

Small suggestion:
I would start using the `stable` branch as the stable docker gets regularly updated with container OS updates automatically while the `master` only gets updated when I updated the master code.
